fedimint-client/src/lib.rs

```diff
@@ -1258,21 +1258,77 @@ impl TransactionUpdates {
     }
 }
 
+/// Federation config and meta data that can be used to show a preview of the
+/// federation one is about to join or to initialize a client.
+#[derive(Debug, Clone)]
+pub struct FederationInfo {
+    config: ClientConfig,
+    // TODO: make non-optional or remove
+    invite_code: Option<InviteCode>,
+}
+
+impl FederationInfo {
+    /// Download federation info using invitation code
+    pub async fn from_invite_code(invite: InviteCode) -> anyhow::Result<FederationInfo> {
+        let config = try_download_config(invite.clone(), 10).await?;
```

Comment:
I just kept the old semantics, but maybe we should make the retry-behavior configurable?

Reply:
I previously thought the client builder would take config for retries. Makes sense for me

Reply:
https://github.com/fedimint/fedimint/issues/3519